src/common/btree.c

```diff
@@ -392,6 +400,9 @@ btr_hkey_size_const(btr_ops_t *ops, uint64_t feats)
 {
 	int size;
 
+	if (BTR_IS_DIRECT_KEY(feats) && BTR_IS_CSUM_ENABLED(feats))
+		return sizeof(umem_off_t) + ops->to_hkey_size();
```

Comment:
you could avoid checking BTR_IS_DIRECT_KEY twice

if (BTR_IS_DIRECT_KEY(feats)) {
     if (BTR_IS_CSUM_ENABLED(feats))
           return ....
    else
          return ...
}

Reply:
changed